src/core/simulation_results.jl

```diff
@@ -218,16 +216,17 @@ function _process_timestamps(
 end
 
 function _get_variables_values(res::SimulationResults, names::Vector{Symbol}, timestamps)
-    _validate_names(res, get_existing_variables, names)
-    same_time_stamps = isempty(setdiff(res.results_timestamps, timestamps))
     existing_names = get_existing_variables(res)
-    same_names = isempty([n for n in names if n ∉ existing_names])
-    if !same_time_stamps && !same_names
-        @debug "reading variables from data store"
-        vals = _get_store_value(res, STORE_CONTAINER_VARIABLES, names, timestamps)
+    _validate_names(existing_names, names)
```

Comment:
You don't really need `get_existing_variables` to return a Vector. It can return an iterator. If anything, `validate_names` would be faster if you passed `existing_names` as a set.

Reply:
want it to return a vector for the user to "see it" if needed. So they can check what's available when interfacing with the SimulationsResults